test/unit/snowflake_test.js

```diff
@@ -1757,6 +1758,32 @@ describe('snowflake.createConnection() JS_TREAT_INTEGER_AS_BIGINT', function ()
   });
 });
 
+
+describe('snowflake.createConnection() VALIDATE_DEFAULT_PARAMETERS', function ()
```

Comment:
can you add test cases that raises the validation error and catch it?
You just need to pass a wrong database, schema or warehouse names and check the error code.